mozldap.go

```diff
@@ -31,26 +31,17 @@ func (cli *Client) Close() {
 // default are used (tls1.2, secure verify, ...).
 //
 // * uri is a connection string to the ldap server, eg. `ldaps://example.net:636/dc=example,dc=net`
-//
 // * username is a bind user, eg. `uid=bind-bob,ou=logins,dc=mozilla`
-//
 // * password is a password for the bind user
-//
 // * cacertpath is the path to a file containing trusted root certificates
-//
 // * tlsconf is a Go TLS Configuration
-//
 // * starttls requires that the LDAP connection is opened insecurely but immediately switched to TLS using the StartTLS protocol.
-func NewClient(uri, username, password string, tlsconf *tls.Config, starttls bool) (cli Client, err error) {
-	defer func() {
-		if e := recover(); e != nil {
-			err = fmt.Errorf("mozldap.NewClient(uri=%q, username=%q, password=****, starttls=%v) -> %v",
-				uri, username, starttls, e)
-		}
-	}()
-	cli, err = ParseUri(uri)
+func NewClient(uri, username, password string, tlsconf *tls.Config, starttls bool) (Client, error) {
+	errorText := fmt.Sprintf("mozldap.NewClient(uri=%q, username=%q, password=****, starttls=%v)", uri, username, starttls)
+
+	cli, err := ParseUri(uri)
 	if err != nil {
-		panic(err)
+		return Client{}, fmt.Errorf("%s -> %s", errorText, err.Error())
```

Comment:
Since we're going to be using a bunch of `fmt.Errorf("%s -> %s", errorText, err.Error())`, would it be useful to have a helper for it? `makeErr(errorText, err)` maybe?
I'm actually not sure if that's any better. Maybe an error type like you mentioned would be a better option: `errorText.New(err)`.